parser.go

```diff
@@ -0,0 +1,140 @@
+package nmea
+
+import (
+	"fmt"
+	"strconv"
+)
+
+// parser provides a simple way of accessing and parsing
+// sentence fields
+type parser struct {
+	Sent
+	prefix string
+	err    error
+}
+
+// newParser constructor
+func newParser(s Sent, prefix string) *parser {
+	p := &parser{Sent: s, prefix: prefix}
+	if p.Type != prefix {
+		p.SetErr("prefix", p.Type)
+	}
+	return p
+}
+
+// Err returns the first error encounterd during the parser's usage.
+func (p *parser) Err() error {
+	return p.err
+}
+
+// SetErr assigns the an error. Calling this method has no
```

Comment:
Fix typo

Reply:
@bogdanprodanj I doubt this is going to get merged.